crates/primitives/src/transaction/mod.rs

```diff
@@ -1304,7 +1329,7 @@ impl TransactionSigned {
 
         let tx_length = header.payload_length + header.length();
         let hash = keccak256(&original_encoding[..tx_length]);
-        Ok((transaction, hash, signature))
+        Ok((transaction, hash, signature.with_parity_bool()))
```

Comment:
hmm, why do we need `.with_parity_bool` here?

Reply:
You're right, this was incorrect.